airflow/utils/dag_processing.py

```diff
@@ -1057,11 +1057,15 @@ def prepare_file_path_queue(self):
             else:
                 file_paths.append(file_path)
 
-            # Find file paths that were recently processed
+            # Find file paths that were recently processed to exclude them
+            # from being added to file_path_queue
+            # unless they were modified recently and parsing mode is "modified_time"
             last_finish_time = self.get_last_finish_time(file_path)
+            file_modified_time = timezone.make_aware(datetime.fromtimestamp(files_with_mtime[file_path]))
```

Comment:
Won't this raise a `KeyError` if we aren't in mtime mode?

Reply:
Updated the logic